integration-tests/test_client_options.py

```diff
@@ -17,3 +22,53 @@ def test_set_ansible_host_info(insights_client):
     ret = insights_client.run("--ansible-host=foo.example.com", check=False)
     assert "Could not update Ansible hostname" not in ret.stdout
     assert ret.returncode == 0
+
+
+def test_group(insights_client, tmp_path):
+    """Test --group option
+    This test generates an archive in offline mode and verifies if the group
+    specified in --group was created and packed
+    """
+
+    group_name = f"RHIQE.{uuid.uuid4()}.testing-group"
+    archive_tmp_name = f"test_archive_{uuid.uuid4()}.tar.gz"
+    archive_location = tmp_path / archive_tmp_name
+    tag_file = None
+
+    # Running insights-client in offline mode to generate archive
+    insights_client.run(
+        "--offline", f"--group={group_name}", f"--output-file={archive_location}"
+    )
+
+    with tarfile.open(archive_location, "r") as tar:
+        for m in tar.getmembers():
+            insights_archive_name = os.path.dirname(m.name)
+            if m.name == f"{insights_archive_name}/tags.json":
+                tag_file = tar.extractfile(m.name)
+                content = tag_file.read()
+                break
+
+    assert tag_file is not None, "Tag file not found"
+
+    tag_file_content = json.loads(content.decode())
```

Comment:
... so here you can pass `tag_file` directly to `json.load()`, which accepts an object on which you can `read()`; this way, you don't even need `decode()`

Reply:
to ensure that you can still use `tag_file` outside of the `for` loop: a possible idea here can be to move the json loading inside the context manager that opens the tarfile, ie
```python
    with tarfile.open(...) as tar:
        for m in tar.getmembers():
            ...
        assert tag_file
        content = json.load(tag_file)
```